src/styles/index.scss

```diff
@@ -1,3 +1,111 @@
-body {
-  margin: 0;
+$date-size: 98px;
+$border-width: 1px;
+$date-bg-color: #eee;
+$hover-color: #ffbfcb;
+$font-size: 30px;
+$gap-size: 1px;
+$padding-size: 10px;
+$hover-offset: -20px;
+$transition-duration: 0.5s;
+$columns: 7;
+$date-position: $date-size + 3 * $gap-size;
+$days: (
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+  sun: 6,
+);
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  width: calc((#{$columns} * #{$date-size}) + 114px);
+  padding: $padding-size;
+  box-sizing: border-box;
+  counter-reset: day;
+  margin-left: 141px;
+  margin-top: 72px;
+
+  @for $i from 1 through 31 {
+    .calendar__day:nth-child(#{$i})::before {
+      content: '#{$i}';
+    }
+  }
+
+  .calendar__day {
+    width: $date-size;
+    height: $date-size;
+    background-color: $date-bg-color;
+    border: $border-width solid black;
+    position: relative;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    font-family: Arial, sans-serif;
+    font-size: $font-size;
+    transition: all $transition-duration;
+
+    &:hover {
+      cursor: pointer;
+      background-color: $hover-color;
+      transform: translateY($hover-offset);
+    }
+
+    &::before {
+      content: counter(date);
+      counter-increment: date;
+    }
+  }
+}
+
+@each $day, $index in $days {
+  $margin: $index * $date-position;
+
+  .calendar--start-day-#{$day} .calendar__day:nth-child(1) {
+    margin-left: $margin;
+  }
+}
```

Comment:
you can nest it inside calendar and use :first-child

Reply:
if i do this code cant work